content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx

```diff
@@ -0,0 +1,530 @@
+
+```ts
+
+const order = await prisma.order.create({
+  data: {
+    // Normal relation
+    product: { connect: { ... } }
+    color: "Red",
+    size: "Large",
+    // Composite type
+    shippingAddress: {
+      {
+        street: "1084 Candycane Lane",
+        city: "Silverlake",
+        zip: "84323"
+      }
+    }
+  }
+})
+
+```
+
+For an optional type, like the `billingAddress`, you can also set the value to `null`:
+
+```ts
+const order = await prisma.order.create({
+  data: {
+    // Embedded optional type, set to null
+    billingAddress: {
+      set: null,
+    },
+  },
+})
+```
+
+To model the case where an `product` contains a list of multiple `photos`, you can `set` multiple composite types at once:
+
+```ts
+const product = prisma.product.create({
+  data: {
+    name: "Forest Runners",
+    price: 59.99
+    // Scalar lists that we already support
+    colors: ["Red", "Green"],
+    sizes: ["Small", "Medium", "Large"]
+    // New composite type
+    photos: {
+      set: [
+				{ height: 100, width: 200, url: "1.jpg" },
+				{ height: 100, width: 200, url: "2.jpg" }
+      ]
+    }
+  }
+})
+```
+
+These operations also work within the `createMany` method. For example, you can create multiple `product`s which each contain a list of `photos`:
+
+```ts
+
+const product = prisma.product.createMany({
+  data: [
+    {
+      name: "Forest Runners",
+      price: 59.99
+      // Scalar lists that we already support
+      colors: ["Red", "Green"],
+      sizes: ["Small", "Medium", "Large"]
+      // New Composite type
+      photos: {
+        set: [
+				  { height: 100, width: 200, url: "1.jpg" },
+				  { height: 100, width: 200, url: "2.jpg" }
+        ]
+      }
+    },
+    {
+      name: "Alpine Blazers",
+      price: 85.99
+      // Scalar lists that we already support
+      colors: ["Blue", "Red"],
+      sizes: ["Large", "XLarge"]
+      // New composite type
+      photos: {
+        set: [
+				  { height: 100, width: 200, url: "1.jpg" },
+				  { height: 150, width: 200, url: "4.jpg" }
+				  { height: 200, width: 200, url: "5.jpg" }
+        ]
+      }
+    }
+  ]
+})
+```
+
+## Changing composite types within <inlinecode>update</inlinecode> and <inlinecode>updateMany</inlinecode>
+
+Composite types can be set, updated or removed within a `update` or `updateMany` method. The following section describes the methods available for updating a single type or multiple types at once, and gives examples of each.
+
+### Changing a single composite type
+
+Use the new `set`, `unset` `update` and `upsert` methods to change a single composite type:
+
+- Use `set` to set a composite type, overriding any existing value
+- Use `unset` to unset a composite type. Unlike `set: null`, `unset` removes the field entirely.
+- Use `update` to update a composite type
+- Use `upsert` to update an existing composite type if it exists, and otherwise create the composite type
```

Comment:
Perhaps break it down to the underlying operations

```suggestion
- Use `upsert` to `update` an existing composite type if it exists, and otherwise `set` the composite type
```

Reply:
Done